mongorestore/restore.go

```diff
@@ -331,11 +332,29 @@ func (restore *MongoRestore) RestoreIntent(intent *intents.Intent) Result {
 	return result
 }
 
-func stripInvalidOptions(indexes []IndexDocument) []IndexDocument {
+func convertLegacyIndexes(indexes []IndexDocument) []IndexDocument {
 	for i, index := range indexes {
+		for j, elem := range index.Key {
```

Comment:
Please factor this out into a `convertLegacyIndexKey` (and later `convertLegacyIndexOptions`).  Otherwise, code complexity is high, hurting readability.  Consider how easy this is to follow (also added an error return -- maybe worth considering):

```
func convertLegacyIndexes(indexes []IndexDocument) ([]IndexDocument, error) {
    var err error
    for i, index := range indexes {
        indexes[i], err = convertLegacyIndexKeys(index)
        if err != nil {
             return nil, err
        }
        indexes[i], err = convertLegacyIndexOptions(index)
        if err != nil {
             return nil, err
        }
    }
    return indexes, nil
}